hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java

```diff
@@ -310,9 +310,11 @@ private static OMClientRequest getOMAclRequest(OMRequest omRequest) {
    * @param exception
    * @return OzoneManagerProtocolProtos.Status
    */
-  public static Status exceptionToResponseStatus(IOException exception) {
+  public static Status exceptionToResponseStatus(Exception exception) {
     if (exception instanceof OMException) {
       return Status.values()[((OMException) exception).getResult().ordinal()];
+    } else if (exception instanceof IllegalArgumentException) {
```

Comment:
IMHO instead of changing here, the validateAndUpdateCache should not throw IllegalArgumentException and throw an OMException and set that as response status using createErrorOMResponse(omResponse, exception)

In general validateAndUpdateCache should not throw any RunTimeExceptions, and it should handle errors and generate response.

Reply:
To explain more we should throw OMException with Status code QUOTA_EXCEEDED, instead of 

throw new IllegalArgumentException("Total buckets quota in this volume " +
          "should not be greater than volume quota : the total space quota is" +
          " set to:" + totalBucketQuota + ". But the volume space quota is:" +
          volumeQuotaInBytes); 

We should use OMException here. Like how we handle any other error related to request logic.

Reply:
yea, make sense, i will do this in a new commit